File: backend/src/models/Referral.js
const mongoose = require('mongoose');

const referralSchema = new mongoose.Schema(
    {
        patientId: {
            type: mongoose.Schema.Types.ObjectId,
            ref: 'Patient',
            required: true,
            index: true,
        },
        fromFacility: { type: String, required: true },
        toFacility: { type: String, required: true },
        referringDoctor: { type: String, required: true },
        specialtyRequired: { type: String, required: true },
        reason: { type: String, required: true },
        priority: {
            type: String,
            enum: ['ROUTINE', 'URGENT', 'EMERGENCY'],
            default: 'ROUTINE',
        },
        status: {
            type: String,
            enum: ['INITIATED', 'ACCEPTED', 'APPOINTMENT_SCHEDULED', 'COMPLETED', 'CANCELLED'],
            default: 'INITIATED',
        },
        referralDate: { type: Date, default: Date.now },
        appointmentDate: { type: Date },
        transportAssistance: { type: Boolean, default: false },
        notes: { type: String, default: '' },
    },
    {
        timestamps: true,
        collection: 'referrals',
    }
);

module.exports = mongoose.model('Referral', referralSchema);

File: lib/core/models/referral.dart
// Referral model — represents a patient referral between healthcare facilities
class Referral {
  final String id;
  final String referredTo;
  final String speciality;
  final String date;
  final String reason;
  final String status; // Pending / Accepted / Completed

  const Referral({
    required this.id,
    required this.referredTo,
    required this.speciality,
    required this.date,
    required this.reason,
    required this.status,
  });

  factory Referral.fromJson(Map<String, dynamic> json) {
    return Referral(
      id: json['id'] as String? ?? '',
      referredTo: json['referredTo'] as String? ?? json['toFacility'] as String? ?? '',
      speciality: json['speciality'] as String? ?? json['specialtyRequired'] as String? ?? '',
      date: json['date'] as String? ?? json['referralDate'] as String? ?? '',
      reason: json['reason'] as String? ?? '',
      status: json['status'] as String? ?? 'Initiated',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'referredTo': referredTo,
      'speciality': speciality,
      'date': date,
      'reason': reason,
      'status': status,
    };
  }
}
